solr/core/src/java/org/apache/solr/search/QueryUtils.java

```diff
@@ -151,7 +151,7 @@ public static BooleanQuery build(BooleanQuery.Builder builder, QParser parser) {
    * If neither are null then we combine with a BooleanQuery.
    */
   public static Query combineQueryAndFilter(Query scoreQuery, Query filterQuery) {
-    if (scoreQuery == null) {
+    if (scoreQuery == null || scoreQuery instanceof MatchAllDocsQuery) {
```

Comment:
I'll add a comment:
`// check for *:* is simple and avoids needless BooleanQuery wrapper even though BQ.rewrite optimizes this away

Reply:
An alternative to a comment could be to include it in the method's javadocs, though perhaps it's more of an implementation detail thingy.

edge case: if `scoreQuery is a MatchAllDocsQuery (which implies it's non-null)` and `filterQuery is null` then would it make sense to return `scoreQuery` itself rather than the current `return new MatchAllDocsQuery();` at line 156?

Reply:
Yes I felt this was an internal detail not worth mentioning.  I pondered how I would document it but felt the javadocs of this method are eclipsing the relative simplicity of what the code does.  I don't think it's worth bothering RE matchAllDocsQuery instantiation.  Really there should be a static singleton for MADQ and when that happens this'd be a non-issue.  It's so minor.